### ISSUE TITLE
Playback failed

### ISSUE THREAD

**Issue by michaelsund** [mazkolain-zz/spotimc#23]
Trying to play any song results in the message "Playback failed, one or more items failed to play"
Heres the logfile: http://pastebin.com/xGPvnpj7

This is not a xbmc-live install, ive used ubuntu-server 10.04.4 X64 running Eden.
I did see the line "ERROR: Unable to open audio mixer" but sound works fine for all other plugins and video/music in  xbmc through the hdmi, ive tried to change some audio settings too, but without any success.

Can i  print out the mixer it wants to use with a testscript?

**Comment by mazkolain-zz** [comment]
Spotimc depends on an embedded proxy server that acts as a bridge between XBMC and the spotify API, serving track and image data through http.

On the log file you provided, at line 12 says that the server is listening on port 8080, and at line 14 you can see how it fails to start, telling that it was unable to open the listening socket.

The proxy will try to listen on the first free port between 8080 and 8089, so I can only think about these two reasons:
1. Theres's something wrong on the free port check.
2. Something at system level is preventing the socket creation (some security policy, etc.).

Anyway, the addon should check better for that kind of situation (seems that the port check wasn't enough). I'll try to reproduce the problem.

Thanks for reporting!

**Comment by michaelsund** [comment]
Fixed! looking at the Edens live cd, there was a python.2.7.X+ package, which probably contains cherrypy3
All i needed was to install python-cherrypy3 and the server started up :)

I did run in to a problem related to the code though, cherrypy does not choose another port than the default 8080.
Since xbmc's webserver runs on 8080 aswell, i changed xbmc's rather than fiddling with your code.
Allthough setting it to 8081 or something could be done with
cherrypy.config.update({'server.socket_port': 8081})
